core/src/main/java/com/yahoo/imapnio/async/request/OpenFolderActionCommand.java

```diff
@@ -0,0 +1,101 @@
+package com.yahoo.imapnio.async.request;
+
+import java.nio.charset.StandardCharsets;
+
+import javax.annotation.Nonnull;
+
+import com.sun.mail.imap.protocol.BASE64MailboxEncoder;
+import com.yahoo.imapnio.async.data.MessageNumberSet;
+import com.yahoo.imapnio.async.data.QResyncParameter;
+import com.yahoo.imapnio.async.exception.ImapAsyncClientException;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+
+/**
+ * This class defines imap abstract commands related to open operation on folder, like select and examine folder.
+ */
+abstract class OpenFolderActionCommand extends ImapRequestAdapter {
+
+    /** Byte array for CR and LF, keeping the array local so it cannot be modified by others. */
+    private static final byte[] CRLF_B = { '\r', '\n' };
+
+    /** Command operator, for example, "SELECT". */
+    private String op;
+
+    /** Folder name. */
+    private String folderName;
+
+    /** Optional QResync parameter. */
+    private QResyncParameter qResyncParameter;
+
+    /**
+     * Initializes a @{code OpenFolderActionCommand}.
+     *
+     * @param op command operator
+     * @param folderName folder name
+     */
+    protected OpenFolderActionCommand(@Nonnull final String op, @Nonnull final String folderName) {
+        this.op = op;
+        this.folderName = folderName;
+        this.qResyncParameter = null;
+    }
+
+    /**
+     * Initializes a @{code OpenFolderActionCommand}.
+     *
+     * @param op command operator
+     * @param folderName folder name to select
+     * @param qResyncParameter qresync parameter
+     */
+    public OpenFolderActionCommand(@Nonnull final String op, @Nonnull final String folderName, @Nonnull final QResyncParameter qResyncParameter) {
+        this.op = op;
+        this.folderName = folderName;
+        this.qResyncParameter = qResyncParameter;
+    }
+
+    @Override
+    public void cleanup() {
+        this.op = null;
+        this.folderName = null;
+        this.qResyncParameter = null;
+    }
+
+    @Override
+    public ByteBuf getCommandLineBytes() throws ImapAsyncClientException {
+        final String base64Folder = BASE64MailboxEncoder.encode(folderName);
+        StringBuilder sb = new StringBuilder();
```

Comment:
Since seqMatchData can contain list of UIDs/sequence number i don't know the size. 
I can remove the size size or put some constant value which might not be useful.